src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -39,5 +41,114 @@ public static IEndpointConventionBuilder RequireHost(this IEndpointConventionBui
             });
             return builder;
         }
+
+        /// <summary>
+        /// Sets the <see cref="EndpointBuilder.DisplayName"/> to the provided <paramref name="displayName"/> for all
+        /// builders created by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointConventionBuilder"/>.</param>
+        /// <param name="displayName">The display name.</param>
+        /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
+        public static IEndpointConventionBuilder WithDisplayName(this IEndpointConventionBuilder builder, string displayName)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Add(b =>
+            {
+                b.DisplayName = displayName;
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="EndpointBuilder.DisplayName"/> using the provided <paramref name="func"/> for all
+        /// builders created by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointConventionBuilder"/>.</param>
+        /// <param name="func">A delegate that produces the display name for each <see cref="EndpointBuilder"/>.</param>
+        /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
+        public static IEndpointConventionBuilder WithDisplayName(this IEndpointConventionBuilder builder, Func<EndpointBuilder, string> func)
```

Comment:
I brought it up as a thing to think about but this extension method feels nieche. If someone really wants to see the display name they could just use:

```cs
builder.Add(b =>
{
    b.DisplayName = "My display name is " + GetDataFromEndpointBuilder(b);
});
```

Reply:
Nah it's not. I literally merged a PR from someone that added a displayname overload to health checks. (which I'm undoing now, sorry Chris)

Reply:
I think it's reasonable for us to have extension methods for the properties of `EndpointBuilder` because they are easy to use.

Reply:
I'm pro the display name setting extension method:
`WithDisplayName(string)` 👍 

I'm on the fence of the utility of the func version: 
`WithDisplayName(Func<EndpointBuilder, string>)` 😐

Reply:
This is to make it obvious that it's possible to do when a convention builder has multiple endpoints. I thought we discussed that.

Reply:
Plus, this helps everyone know that we're smart.

Reply:
We did. I'm not sold on what has resulted from bringing it up 🙃 

Should there be func versions of the add metadata extension methods?

Reply:
No, I don't think so because we haven't identified a clear use case for metadata. 

Prior art here MVC + route name + token replacement.

If you have something that you really want to be unique like a name, it helps to give people a reasonable way to set it.